### PR TITLE
Zoltan2: Fixes for #6440

### PR DESCRIPTION
**Pull request by MicheldeMessieres**
This will resolve indeterminate behavior in the Multijagged calculation.
This is expected to resolve most of the cases where 6440 failed. However in one case, cdash showed a segfault and it's not clear yet why that would occur or if it would be impacted by this fix. This will at least get the test running in a consistent manner.

@trilinos/zoltan2

## Motivation

Eliminates indeterminate behavior. Will fix most but perhaps not all intermittent failures reported in #6440.

### CODE REVIEW

File: packages/zoltan2/src/algorithms/partition/Zoltan2_AlgMultiJagged.hpp [hunk 1]
```diff
@@ -4829,6 +4830,12 @@ mj_create_new_partitions(
   delete [] reduce_array;
 #endif
 
+  // the last member is utility used for atomically inserting the values.
+  // Sorting here avoids potential indeterminancy in the partitioning results
+  auto track_on_cuts_sort = Kokkos::subview(track_on_cuts,
+    std::pair<mj_lno_t, mj_lno_t>(0, track_on_cuts.size() - 1));
```
Comment (mhoemmen) [inline packages/zoltan2/src/algorithms/partition/Zoltan2_AlgMultiJagged.hpp]: `Kokkos::subview` takes a half-exclusive range `(start, end+1)`.  Also, what if `track_on_cuts.size()` is zero?

**Comment by MicheldeMessieres** [comment]
@mhoemmen Thanks. This array is always size 1 or greater and the last element is used for index tracking so not sorted. Though I could add a check to skip the sort when size = 1 and no sorting is necessary. However I've temporarily closed the PR as the bug just stopped replicating for me in develop.  I need to investigate this further.

**Comment by MicheldeMessieres** [comment]
@kddevin PR fixes indeterminate result causing the MultiVector pass and BasicVectorAdapter pass to give different results on the cuts and fail the comparison part of the test.

I've also fixed another issue in the GeometricGenerator random number generation. The statics were not handled in a thread safe way and the original coordinate distribution was changing run to run for OpenMP. I think performance for GeometricGenerator is not very important since it's for tests only so I've simplified this and just removed the statics.

**Comment by kddevin** [comment]
Thanks, @MicheldeMessieres.   
You are correct that performance of the GeometricGenerator does not matter.  If its use of OpenMP is problematic, it is OK to serialize it all.

**Pull request merged**